File: download_images.py
import os
import urllib.request
import time

base_dir = r"e:\appartmentservice\images"
categories = ["All", "Building", "Living Room", "Dining & Kitchen", "Bedroom", "Bathroom", "Laundry"]

if not os.path.exists(base_dir):
    os.makedirs(base_dir)

for i, category in enumerate(categories):
    cat_dir = os.path.join(base_dir, category)
    if not os.path.exists(cat_dir):
        os.makedirs(cat_dir)
    
    for j in range(1, 9):
        seed = f"apartmentservice_{i}_{j}"
        url = f"https://picsum.photos/seed/{seed}/800/600"
        
        filename = f"{category.replace(' ', '_').replace('&', 'and').lower()}_{j}.jpg"
        filepath = os.path.join(cat_dir, filename)
        
        try:
            req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
            with urllib.request.urlopen(req) as response, open(filepath, 'wb') as out_file:
                out_file.write(response.read())
            print(f"Downloaded {filepath}")
        except Exception as e:
            print(f"Failed to download {filepath}: {e}")
        time.sleep(0.2)

print("Images generated successfully.")

File: download_images.js
const https = require('https');
const fs = require('fs');
const path = require('path');

const baseDir = path.join(__dirname, 'images');
const categories = ["All", "Building", "Living Room", "Dining & Kitchen", "Bedroom", "Bathroom", "Laundry"];

if (!fs.existsSync(baseDir)) {
    fs.mkdirSync(baseDir);
}

const downloadImage = (url, filepath) => {
    return new Promise((resolve, reject) => {
        https.get(url, (response) => {
            if (response.statusCode >= 300 && response.statusCode < 400 && response.headers.location) {
                // follow redirect
                downloadImage(response.headers.location, filepath).then(resolve).catch(reject);
                return;
            }
            if (response.statusCode !== 200) {
                return reject(new Error(`Failed to get '${url}' (${response.statusCode})`));
            }
            const file = fs.createWriteStream(filepath);
            response.pipe(file);
            file.on('finish', () => {
                file.close(resolve);
            });
        }).on('error', (err) => {
            fs.unlink(filepath, () => reject(err));
        });
    });
};

async function main() {
    for (let i = 0; i < categories.length; i++) {
        const category = categories[i];
        const catDir = path.join(baseDir, category);
        
        if (!fs.existsSync(catDir)) {
            fs.mkdirSync(catDir);
        }
        
        for (let j = 1; j <= 8; j++) {
            const seed = `apartmentservice_${i}_${j}`;
            const url = `https://picsum.photos/seed/${seed}/800/600`;
            
            const filename = `${category.replace(/\s+/g, '_').replace(/&/g, 'and').toLowerCase()}_${j}.jpg`;
            const filepath = path.join(catDir, filename);
            
            try {
                await downloadImage(url, filepath);
                console.log(`Downloaded ${filepath}`);
            } catch (err) {
                console.error(`Failed to download ${filepath}: ${err.message}`);
            }
        }
    }
    console.log("Images generated successfully.");
}

main();
